src/devices/video/mc6847.cpp

```diff
@@ -559,11 +588,23 @@ void mc6847_base_device::device_config_complete()
 
 	if (!screen().refresh_attoseconds())
 	{
-		// FIXME: use correct raw parameters rather than this nonsense
-		screen().set_refresh_hz(m_tpfs > 310.0 ? 50 : 60);
-		screen().set_size(320, 243);
-		screen().set_visarea(0, 320-1, 1, 241-1);
-		screen().set_vblank_time(0);
+		// It is preferred to initialize the screen device using set_raw.
+		// Setting screen device's raw parameters requires values in terms of pixels.
+		// Multiplying the 6847's clock by 2 produces a pixel clock that allows
+		// 256 pixels of active video area per line, with apparently no roundoff
+		// error or drift.
+		//
+		// Note: This math is valid for NTSC.  If / when PAL is implemented, m_tpfs
+		// can be used to determine if this 6847 is PAL, and a different set_raw
+		// call can be made if helpful.
```

Comment:
I wonder what the real 6847 really do in order to pinpoint that is a PAL machine?
From quick glance from the datasheets the VDG seems coupled with a MC1372, which is what drives the clock on a NTSC machine, no idea about PAL atm but regardless I guess that `m_tpfs` is some kind of cheap simplification that should be removed in the end.

Reply:
Yeah, I wish I had a good answer to that, but I'm brand new to all of this (hardware and MAME).  The original MC6847.cpp code switched off of m_tpfs when setting the refresh rate.  I wanted to preserve that approach in a comment when I changed the code to use set_raw, so that future changes will at least have a starting point to consider.